### ISSUE TITLE
Pipeline UI react errors on keydown

### ISSUE THREAD

**Issue by jhendley25** [spinnaker/spinnaker#5279]
### Issue Summary:
There's a react error happening on keydown when adding an automated trigger, causing the page to rerender after every key press. Minified code points to this error: https://reactjs.org/docs/error-decoder.html/?invariant=31&args[]=object%20with%20keys%20%7Ba%7D&args[]=

### Cloud Provider(s):

GCP

### Environment:
Distributed kubernetes

### Feature Area:
Pipelines UI, specifically the inputs in the Automated Triggers section

### Description:
There's a react error happening on keydown when adding an automated trigger

### Steps to Reproduce:
Try to add payload constraints, everything will re-render and focus is lost on the input.

### Additional Details:
![image](https://user-images.githubusercontent.com/5239363/71725551-648d1000-2e02-11ea-8413-a6d6e3fb652a.png)

**Comment by maggieneterval** [comment]
Hi @jhendley25, thanks for taking the time to open this issue. A couple of questions to help us get to the bottom of this:

- What version of Spinnaker are you running?
- What trigger type's payload constraints are you attempting to edit? Both Webhook and PubSub have payload constraints, and I am not able to reproduce this issue at HEAD while editing either type. Are you able to attach a gif of making the edits in the UI that cause the error?
- Do you have the [`artifactsRewrite`](https://www.spinnaker.io/reference/artifacts-with-artifactsrewrite/) feature flag enabled?

Thanks!

**Comment by christopherthielen** [comment]
I manually keyed in the URL https://reactjs.org/docs/error-decoder.html/?invariant=31&args[]=object%20with and the error message is: 

```
Objects are not valid as a React child (found: ... rest is missing
```

**Comment by jhendley25** [comment]
Hey @maggieneterval answers below:

Version: 1.17.5

For trigger types, I just realized that it was only happening in the "Key" field of both inputs in the screenshot below:

![image](https://user-images.githubusercontent.com/5239363/71844202-ad0f2c80-3093-11ea-855e-eb755616cd12.png)

No feature flags enabled that I'm aware of. I managed to get around the issue by copy/pasting the strings into the fields.

**Comment by lukmy** [comment]
For my situation this error occured while pressing `add payload constraint` @maggieneterval 

My env lists below: 
- 1.17.5, 
- spinnaker: distributed kubernetes
- cloud provider: kubernetes
- artifactsRewrite: whether enabled or disabled, this error still occurs

**Comment by maggieneterval** [comment]
Thanks for the additional info, @jhendley25 and @lukmy. It looks like the wonderful @christopherthielen already merged [a fix](https://github.com/spinnaker/deck/pull/7601) for this bug, but we never cherry-picked it into the 1.17 release branch. Unless @christopherthielen has any concerns with cherry-picking those commits, we can go ahead and do that now so this will be fixed with the next patch release of 1.17. (1.18 will also be released within the next week or so and will definitely include these fixes).
